src/http/content.cr

```diff
@@ -61,7 +61,7 @@ module HTTP
     getter headers : HTTP::Headers { HTTP::Headers.new }
 
     def initialize(@io : IO)
-      @chunk_remaining = 0
+      @chunk_remaining = -1
```

Comment:
maybe use `nil` instead as an "invalid value" ?

Reply:
It's only an internal variable so it should be fine to use a special value to determine initial state.
If this should be changed, it would probably make more sense to add an additional ivar like `@initial_read`. That would be more explicit. Not that it matters much, but making it nilable would add 12 bytes, a Bool ivar only 1.

Reply:
more like 8 bytes for the boolean once you align. We should do the ivar reordering/packing that rust does eventually...